File: src/dlp/ast/expressions/StructAccess.ts

import {Expression} from "./Expression";
import {AbstractLocatable} from "../AbstractLocatable";

export class ArrayAccess extends AbstractLocatable implements Expression {

    name: String;
    value: Expression;

    constructor( name: String, value: Expression, line: number, column: number){
        super(line, column);
        this.name = name;
        this.value = value;
    }
}

File: src/ast/expressions/StructAccess.java
package ast.expressions;

import ast.AbstractLocatable;

public class StructAccess extends AbstractLocatable implements Expression {

    private String name;
    private Expression value;

    public StructAccess(String name, Expression value, int line, int column) {
        super(line, column);
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Expression getValue() {
        return value;
    }
}
